src/opm/common/utility/TimeService.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+  Copyright 2019 Equinor ASA.
+
+  This file is part of the Open Porous Media Project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/common/utility/TimeService.hpp>
+
+#include <chrono>
+#include <ctime>
+#include <utility>
+
+namespace {
+    std::time_t advance(const std::time_t tp, const double sec)
+    {
+        using namespace std::chrono;
+
+        using TP      = time_point<system_clock>;
+        using DoubSec = duration<double, seconds::period>;
+
+        const auto t = system_clock::from_time_t(tp) +
+            duration_cast<TP::duration>(DoubSec(sec));
+
+        return system_clock::to_time_t(t);
+    }
+
+    std::time_t makeUTCTime(const std::tm& timePoint)
+    {
+        auto       tp    =  timePoint; // Mutable copy.
```

Comment:
Instead of taking a local copy you could just take a mutable value in the signature?

Reply:
Good point.  I'll update accordingly.

Reply:
I've made this change.